functions/problem-set-functions.md

```diff
@@ -0,0 +1,243 @@
+
+```js
+const isOdd = num => {
+    return num % 2;
+};
+```
+
+Another example of a working implementation:
+
+```js
+const isOdd = num => num % 2;
+```
+
+### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 3 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: custom-snippet
+* language: javascript
+* id: 0d7fc6c9
+* title: Functions
+* docker_directory_path: /custom-snippets/isOddNumber
+##### !question
+
+Implement the function `isOddNumber`.
+
+`isOddNumber` should take in one number, `num`. It returns a Truthy value if `num` is an odd number. It returns a Falsy value if `num` is an even number.
+
+If `num` is not a number, the function should return a Falsy value.
+
+| `num`   | Expected Return Value |
+| ------- | --------------------- |
+| `1`     | Truthy                |
+| `2`     | Falsy                 |
+| `hello` | Falsy                 |
+| `[]`    | Falsy                 |
+| `{}`    | Falsy                 |
+
+##### !end-question
+### !explanation
+
+- The result of `'hello' % 2` is `NaN`, which is falsy
+- The result of `[] % 2` is `0`, which is falsy
+- The result of `{} % 2` is `NaN`, which is falsy
+
+<br/>
+
+An example of a working implementation:
+
+```js
+const isOddNumber = num => {
+    return num % 2;
+};
+```
+
+Another example of a working implementation:
+
+```js
+const isOddNumber = num => num % 2;
+```
+
+### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 4 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: custom-snippet
+* language: javascript
+* id: d522a65a
+* title: Functions
+* docker_directory_path: /custom-snippets/giveGreetingChallenge
+##### !question
+
+Implement the following three functions:
+
+- `giveCasualGreeting`
+- `giveFormalGreeting`
+- `writeLetter`
+
+`giveCasualGreeting` takes in one parameter, `name`.
+
+`name` | Expected Return Value
+--- | ---
+`'Miranda'` | `'Howdy, Miranda!'`
+`'Jonathan'` | `'Howdy, Jonathan!'`
+
+`givesFormalGreeting` takes in one parameter, `name`.
+
+`name` | Expected Return Value
+--- | ---
+`'Miranda'` | `'Dear Mx. Miranda 🤵'`
+`'Jonathan'` | `'Dear Mx. Jonathan 🤵'`
+
+`writeLetter` takes in _two_ parameters:
+- `giveGreeting`, which is a function
+- `name`
+
+| <div style="width:200px;">`giveGreeting`</div> | <div style="width:125px;">`name`</div> | Expected Return Value
+--- | --- | ---
+`giveCasualGreeting` | `'Miranda'` | `'Howdy, Miranda! I hope this letter finds you well.'`
+`giveCasualGreeting` | `'Jonathan'` | `'Howdy, Jonathan! I hope this letter finds you well.'`
+`giveFormalGreeting` | `'Miranda'` | `'Dear Mx. Miranda 🤵 I hope this letter finds you well.'`
+`giveFormalGreeting` | `'Jonathan'` | `'Dear Mx. Jonathan 🤵 I hope this letter finds you well.'`
+
+##### !end-question
+### !hint
+
+Inside the function `writeLetter`, `giveGreeting` is a function. We invoke this function with `giveGreeting()`. We can pass in a variable into `giveGreeting`:
```

Comment:
minimal fix
```suggestion
Inside the function `writeLetter`, `giveGreeting` is a function. We invoke this function with `giveGreeting()`. We can pass a variable into `giveGreeting`:
```

I prefer reminding the reader that the ability to pass a parameter is only because that's how we expect it to be used.
```suggestion
Inside the function `writeLetter`, `giveGreeting` is a function. We invoke this function with `giveGreeting()`. We expect it to accept a parameter, so we can pass a variable into `giveGreeting`:
```